modules/couchbase/src/main/java/org/testcontainers/containers/CouchbaseContainer.java

```diff
@@ -0,0 +1,284 @@
+        newBuckets.add(bucketSettings);
+        return self();
+    }
+
+
+    public void initCluster() {
+        urlBase = String.format("http://%s:%s", getContainerIpAddress(), getMappedPort(8091));
+        try {
+            String poolURL = "/pools/default";
+            String poolPayload = "memoryQuota=" + URLEncoder.encode(memoryQuota, "UTF-8") + "&indexMemoryQuota=" + URLEncoder.encode(indexMemoryQuota, "UTF-8");
+
+            String setupServicesURL = "/node/controller/setupServices";
+            StringBuilder servicePayloadBuilder = new StringBuilder();
+            if (keyValue) {
+                servicePayloadBuilder.append("kv,");
+            }
+            if (query) {
+                servicePayloadBuilder.append("n1ql,");
+            }
+            if (index) {
+                servicePayloadBuilder.append("index,");
+            }
+            if (fts) {
+                servicePayloadBuilder.append("fts,");
+            }
+            String setupServiceContent = "services=" + URLEncoder.encode(servicePayloadBuilder.toString(), "UTF-8");
+
+            String webSettingsURL = "/settings/web";
+            String webSettingsContent = "username=" + URLEncoder.encode(clusterUsername, "UTF-8") + "&password=" + URLEncoder.encode(clusterPassword, "UTF-8") + "&port=8091";
+
+            String bucketURL = "/sampleBuckets/install";
+
+            StringBuilder sampleBucketPayloadBuilder = new StringBuilder();
+            sampleBucketPayloadBuilder.append('[');
+            if (travelSample) {
+                sampleBucketPayloadBuilder.append("\"travel-sample\",");
+            }
+            if (beerSample) {
+                sampleBucketPayloadBuilder.append("\"beer-sample\",");
+            }
+            if (gamesIMSample) {
+                sampleBucketPayloadBuilder.append("\"gamesim-sample\",");
+            }
+            sampleBucketPayloadBuilder.append(']');
+
+            callCouchbaseRestAPI(poolURL, poolPayload);
+            callCouchbaseRestAPI(setupServicesURL, setupServiceContent);
+            callCouchbaseRestAPI(webSettingsURL, webSettingsContent);
+            callCouchbaseRestAPI(bucketURL, sampleBucketPayloadBuilder.toString());
+
+            CouchbaseWaitStrategy s = new CouchbaseWaitStrategy();
+            s.withBasicCredentials(clusterUsername, clusterPassword);
+            s.waitUntilReady(this);
+            callCouchbaseRestAPI("/settings/indexes", "indexerThreads=0&logLevel=info&maxRollbackPoints=5&storageMode=memory_optimized");
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void createBucket(BucketSettings bucketSetting, Boolean createIndex){
+        BucketSettings bucketSettings = getCouchbaseCluster().clusterManager(clusterUsername, clusterPassword).insertBucket(bucketSetting);
+        // allow some time for the query service to come up
+        try {
+            Thread.sleep(5000);
```

Comment:
http://docs.couchbase.com/admin/admin/CLI/CBcli/cbcli-bucket-create.html
I see `--wait` option here - how does it wait?

Also, at least something like `Unreliables.retryUntilSuccess(1, TimeUnit.MINUTES, () -> clusterManager.hasBucket(bucketName))` would be better I think